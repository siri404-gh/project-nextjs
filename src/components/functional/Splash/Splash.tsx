import React, { useEffect, FunctionComponent } from 'react';
import {
  LinearProgress,
  makeStyles,
} from '@material-ui/core';

import splash from '@/assets/splash.png';

export interface Props {
  onLoad?: () => void,
  duration?: number,
}

const useStyles = makeStyles(theme => ({
  splash: {
    position: 'absolute',
    top: 0,
    left: 0,
    display: 'flex',
    flexDirection: 'column',
    minHeight: '-webkit-fill-available',
    width: '100%',
    background: `url(${splash}) center center no-repeat #ffffff`,
    backgroundSize: 'contain',
  },
  linearProgress: {
    width: '100%',
    position: 'absolute',
    top: 0,
  },
}));

const Splash: FunctionComponent<Props> = ({ onLoad, duration }) => {

  useEffect(() => {
    const timer = setTimeout(onLoad, duration);
    return () => clearTimeout(timer);
  }, []);

  const classes = useStyles();

  return (
    <div className={classes.splash}>
      <div className={classes.linearProgress}>
        <LinearProgress variant='indeterminate' />
      </div>
    </div>
  );
};

Splash.defaultProps = {
  onLoad: /* istanbul ignore next */ () => null,
  duration: 1000,
};

export default Splash;
