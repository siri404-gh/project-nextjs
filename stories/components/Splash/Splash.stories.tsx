import React from 'react';
import { Story } from '@storybook/react/types-6-0';

import Splash, { Props } from '@/components/functional/Splash/Splash';

export default {
  title: 'Splash',
  component: Splash,
};

const Template: Story<Props> = args => <Splash {...args} />;

export const FirstStory = Template.bind({});

FirstStory.args = {};
