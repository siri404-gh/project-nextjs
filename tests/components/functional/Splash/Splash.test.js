import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import Splash from '@/components/functional/Splash/Splash';

jest.useFakeTimers();

describe('Splash', () => {
  let container = null;

  beforeEach(() => {
    container = document.createElement("section");
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it("renders without props", () => {
    act(() => {
      render(<Splash />, container)
    });
    expect(container).toMatchSnapshot();
  });

  it("renders with props", () => {
    const props = {
      duration: 1000,
      onLoad: jest.fn(),
    }
    act(() => {
      render(<Splash {...props} />, container)
    });
    expect(container).toMatchSnapshot();
  });

  it("calls onLoad after 1 second", () => {
    const props = {
      duration: 1000,
      onLoad: jest.fn(),
    };

    act(() => {
      render(<Splash {...props} />, container);
    });

    act(() => {
      jest.advanceTimersByTime(100);
    });

    expect(props.onLoad).not.toHaveBeenCalled();

    act(() => {
      render(null, container);
    });

    act(() => {
      render(<Splash {...props} />, container);
    });

    act(() => {
      jest.advanceTimersByTime(1000);
    });

    expect(props.onLoad).toHaveBeenCalled();
  });
});
