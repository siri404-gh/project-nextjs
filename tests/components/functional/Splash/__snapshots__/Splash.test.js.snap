// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash renders with props 1`] = `
<section>
  <div
    class="makeStyles-splash-3"
  >
    <div
      class="makeStyles-linearProgress-4"
    >
      <div
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
        role="progressbar"
      >
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
        />
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
        />
      </div>
    </div>
  </div>
</section>
`;

exports[`Splash renders without props 1`] = `
<section>
  <div
    class="makeStyles-splash-1"
  >
    <div
      class="makeStyles-linearProgress-2"
    >
      <div
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
        role="progressbar"
      >
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
        />
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
        />
      </div>
    </div>
  </div>
</section>
`;
